core/src/main/scala/pureconfig/DurationConvert.scala

```diff
@@ -42,9 +43,11 @@ private[pureconfig] object DurationConvert {
    */
   def fromDuration(d: Duration): String = {
     d match {
+      case f: FiniteDuration => fromFiniteDuration(f)
+      // We must do an `eq` instead of `==` comparison because Undefined is intentionally != itself.
+      case i: Duration.Infinite if i eq Duration.Undefined => Duration.Undefined.toString
       case i: Duration.Infinite if i == Duration.MinusInf => "MinusInf"
       case i: Duration.Infinite => "Inf"
```

Comment:
@jcazevedo I don't understand where you see an inconsistency. Are you saying that that the `if eq Duration.Undefined` is inconsistent because it's not `i.toString == Duration.Undefined.toString`?
I agree that the string `"Duration.Undefined"` in a configuration is ugly, but it seems somewhat less likely to be misleading than `undefined` which has the potential to present problems in the future if we ended up with a JSON backend for pureconfig.